packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -314,13 +310,39 @@ function synchronizeDefaultValue(
 ) {
   if (
     // Focused number inputs synchronize on blur. See ChangeEventPlugin.js
-    (type !== 'number' || node.ownerDocument.activeElement !== node) &&
-    node.defaultValue !== '' + value
+    type !== 'number' ||
+    node.ownerDocument.activeElement !== node
   ) {
-    if (value != null) {
-      node.defaultValue = '' + value;
-    } else {
-      node.defaultValue = node._wrapperState.initialValue;
-    }
+    assignProperty(
+      node,
+      'defaultValue',
+      value == null ? node._wrapperState.initialValue : value,
+    );
+  }
+}
+
+function assignProperty(
+  node: InputWithWrapperState,
+  property: string,
+  value: *,
+) {
+  // TODO: This should use DOMProperty.shouldSetAttribute, however the current
+  // behavior is such that invalid attributes do not update the current attribute,
+  // instead of removing it.
+  switch (typeof value) {
+    case 'boolean':
+    case 'number':
+    case 'string':
+    case 'object':
+      // Use hasOwnProperty instead of checking for '' to ensure properties
+      // are set for the first time. This is important for defaultValue, which
+      // otherwise does not set an empty value attribute
+      if (!node.hasOwnProperty(property) || node[property] !== '' + value) {
```

Comment:
We must only compare the stringified `value` property/attribute after we've confirmed that it isn't a symbol. Otherwise you get a run time exception:

<img width="370" alt="screen shot 2017-12-01 at 8 51 40 am" src="https://user-images.githubusercontent.com/590904/33485591-e87b1304-d674-11e7-8f0a-4f3feb26de53.png">

Maybe a safer way to do this would be to wrap the value in a string constructor?

<img width="227" alt="screen shot 2017-12-01 at 8 52 36 am" src="https://user-images.githubusercontent.com/590904/33485608-fddbc77a-d674-11e7-83f2-6f8af1bd8774.png">

<img width="362" alt="screen shot 2017-12-01 at 8 53 02 am" src="https://user-images.githubusercontent.com/590904/33485626-114ff678-d675-11e7-95c0-397194445712.png">

Reply:
AFAIK `String()` is slower, at least in V8.

Reply:
Sure enough:

![Uploading Screen Shot 2017-12-01 at 9.07.22 AM.png…]()

Reply:
Should we add a TODO: to remove this on a major bump? It's not clear to me why ignoring other types vs throwing or defering to whatever the browser does is more corrrect

Reply:
I think the constraint we care about is that objects get stringified. Since we depend on this a lot. Therefore we can't use browser APIs that don't attempt to stringify them.

Reply:
Sure that's sense, tho I think "treating the as null" vs an error is maybe less ideal. In, the error case it's easier track down and fix vs inputs clearing and maybe not noticing

Reply:
I believe the unknown property hook will still raise a warning.